### DISCUSSION TITLE
Installing some libraries

### DISCUSSION THREAD

**Discussion by khalidrabayah** [community/community#34811] [Programming Help] [answered]
I am trying to run the optional labs of the machine learning specialization, however, i stcuk with some libraries and functions that
I can not install like the ones below, 

plt.style.use('./deeplearning.mplstyle')
from lab_utils_uni import plt_house_x, plt_contour_wgrad, plt_divergence, plt_gradients

I got the error message 
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-14-b6cfcb1e5d3a> in <module>
      3 import matplotlib.pyplot as plt
      4 # plt.style.use('./deeplearning.mplstyle')
----> 5 from lab_utils_uni import plt_house_x, plt_contour_wgrad, plt_divergence, plt_gradients

ModuleNotFoundError: No module named 'lab_utils_uni'

**Comment by sethjuarez** [comment]
Welcome and thanks for the question! Looks like there's a missing dependency - were you able to solve this?

**Comment by ademakdogan** [comment, marked as answer]
Thatʻs simple, copy lab_utils_uni.py file to the same directory your code is saved to. You can download that python files from the "files" tab on Coursera online tab.

**Comment by PriyanshNegi** [comment]
It seems like you're encountering an issue with importing certain modules and functions while running the optional labs of the machine learning specialization. The error message suggests that the module 'lab_utils_uni' cannot be found, resulting in a "ModuleNotFoundError".

To resolve this issue, please make sure that you have correctly installed all the required dependencies and libraries for the labs. It's possible that the 'lab_utils_uni' module is not included in the package you have installed or is located in a different directory.

Here are a few steps you can take to address the problem:

1. Double-check the installation instructions: Review the documentation or instructions provided with the labs and ensure that you have followed all the installation steps accurately. Pay special attention to any additional dependencies or specific setup instructions.

2. Confirm the availability of 'lab_utils_uni': Verify if the 'lab_utils_uni' module is indeed a part of the labs or if it needs to be obtained separately. Check the lab materials or any accompanying documentation for information on how to obtain or install the 'lab_utils_uni' module.

3. Search for alternative resources: If you are unable to find the 'lab_utils_uni' module or if it's causing persistent issues, try searching for alternative resources or libraries that provide similar functionality. You can explore popular machine learning libraries like scikit-learn or TensorFlow, which offer a wide range of utility functions for plotting, data manipulation, and model evaluation.

4. Seek assistance from the course or lab community: If the problem persists, consider reaching out to the course or lab community for further assistance. They may have encountered the same issue and can provide specific guidance or alternative solutions.

Troubleshooting specific library or module-related issues can be challenging without detailed knowledge of the lab and its dependencies. It's important to consult the lab documentation, installation guides, and seek support from the relevant community to ensure a smooth learning experience.

**Comment by ali-workshop** [comment]
The error you're encountering, ModuleNotFoundError: No module named 'lab_utils_uni', indicates that Python cannot find the lab_utils_uni module, which is likely a custom utility module provided by the course or lab materials. This issue is not directly related to the plt.style.use('./deeplearning.mplstyle') line, which is about applying a custom style to matplotlib plots. However, addressing the lab_utils_uni issue will likely resolve the overall problem you're facing with the optional labs of the machine learning specialization.

To resolve the ModuleNotFoundError for lab_utils_uni, follow these steps:

Ensure All Files Are Downloaded: Make sure you have downloaded all the necessary files from the course or lab materials. This includes not only the deeplearning.mplstyle file but also any utility modules like lab_utils_uni. These files should be placed in the same directory as your main Python script or notebook to ensure they are accessible [1](https://stackoverflow.com/questions/73943812/plt-style-use-deeplearning-mplstyle-is-not-working).
Check Your Working Directory: Ensure that your Python script or Jupyter notebook is running in the correct directory where all the necessary files are located. You can check your current working directory in Python using import os; print(os.getcwd()).
Verify File Names and Paths: Double-check the file names and paths to ensure they match exactly with what's being imported in your script. Python is case-sensitive, so lab_utils_uni is different from Lab_Utils_Uni.
Update Your Python Environment: If you're using a virtual environment (which is recommended), make sure it's activated and contains all the necessary packages for your project. You can update your environment by running pip install -r requirements.txt if you have a requirements.txt file listing all dependencies.
Consult Course Materials: If the lab_utils_uni module is part of the course materials, consult the course documentation or forums for any specific instructions on setting up the environment or resolving common issues.
